bentoml/marshal/marshal.py

```diff
@@ -157,14 +158,17 @@ def set_outbound_port(self, outbound_port):
 
     def fetch_sema(self):
         if self._outbound_sema is None:
-            self._outbound_sema = asyncio.Semaphore(self.outbound_workers)
+            self._outbound_sema = NonBlockSema(self.outbound_workers)
         return self._outbound_sema
 
     def add_batch_handler(self, api_name, max_latency, max_batch_size):
 
         if api_name not in self.batch_handlers:
             _func = ParadeDispatcher(
-                max_latency, max_batch_size, shared_sema=self.fetch_sema
+                max_latency,
+                max_batch_size,
+                shared_sema=self.fetch_sema(),
+                fallback=aiohttp.web.HTTPTooManyRequests,
```

Comment:
Is there a case when `fallback` is not `aiohttp.web.HTTPTooManyRequests`? maybe just set it within ParadeDispatcher for now? something like `self.fallback = aiohttp.web.HTTPTooManyRequests`

Reply:
Each request need a new aiohttp.web.HTTPTooManyRequests object or cause exceptions.

Reply:
The ParadeDispatcher(or CorkDispatcher ?) is just a universal function-wise lib. It's not coupling with http.

Reply:
make sense